OpenRA.Mods.Common/Scripting/Properties/PlayerProperties.cs

```diff
@@ -74,5 +74,15 @@ public Actor[] GetActorsByType(string type)
 
 			return result.ToArray();
 		}
+
+		[Desc("Check if the player has this prerequisite available.")]
+		public bool HasPrerequisiteAvailable(string type)
+		{
+			var tt = Player.PlayerActor.TraitOrDefault<TechTree>();
+			if (tt == null)
+				throw new LuaException("Missing TechTree trait on player {0}!".F(Player));
+
+			return tt.HasPrerequisites(type.OfType<string>());
```

Comment:
Hmm, this doesn't seem to work. `TechTree.HasPrerequisites` returns `true` for a prerequisite the player doesn't actually have.

Testcase:
```diff
diff --git a/mods/cnc/maps/gdi01/gdi01.lua b/mods/cnc/maps/gdi01/gdi01.lua
index 1855f74..55ad272 100644
--- a/mods/cnc/maps/gdi01/gdi01.lua
+++ b/mods/cnc/maps/gdi01/gdi01.lua
@@ -97,11 +97,17 @@ WorldLoaded = function()
        Trigger.AfterDelay(DateTime.Seconds(60), function() Reinforce(VehicleReinforcements) end)
 end
 
+done = false
 Tick = function()
        if enemy.HasNoRequiredUnits() then
                player.MarkCompletedObjective(secureAreaObjective)
        end
 
+       if not done and player.HasPrerequisiteAvailable("nuke") then
+               Media.DisplayMessage("Power plant built.")
+               done = true
+       end
+
        if DateTime.GameTime > DateTime.Seconds(5) and player.HasNoRequiredUnits() then
                player.MarkFailedObjective(beachheadObjective)
                player.MarkFailedObjective(secureAreaObjective)
```